apps/lib/http_server.c

```diff
@@ -222,7 +222,7 @@ BIO *http_server_init_bio(const char *prog, const char *port)
     int asock;
     char name[40];
 
-    snprintf(name, sizeof(name), "[::]:%s", port); /* port may be "0" */
+    BIO_snprintf(name, sizeof(name), "[::]:%s", port); /* port may be "0" */
```

Comment:
The same code has been reported as causing test failures on systems with IPv6 disabled.
Because `[::]` then does not work. We'd likely get better results from `0.0.0.0` while we're at it.
See [openssl-users](https://groups.google.com/a/openssl.org/g/openssl-users/c/DYfcbaBVzms/m/M05hkgknAQAJ)
which does suggest that additional IPv6-related fixes are required.

Reply:
> We'd likely get better results from `0.0.0.0` while we're at it.

That would break listening for IPv6 connections on IPv6 enabled systems. I do not think this should be changed.

Reply:
Besides it is a different problem to fix so it should be a different PR.